### PR TITLE
chore: fitness quality gate to only allow TS & TSX files in app directory

### PR DESCRIPTION
**Pull request by sethkfman**
## **Description**

This PR adds a fitness function to fail if a .js or .jsx file is added to the app/ directory. 

## **Related issues**

Fixes: https://github.com/orgs/MetaMask/projects/60/views/3?pane=issue&itemId=59597337

## **Manual testing steps**

1. Attempt to push a .js or .jsx file  in the app directory
2. See the fitness function fail
3.

## **Screenshots/Recordings**

NA See unit tests

### **After**

- Commit [outside of app directory](https://github.com/MetaMask/metamask-mobile/actions/runs/9353695806/job/25744808362)
- Commit [inside of app directory](https://github.com/MetaMask/metamask-mobile/actions/runs/9455916396/job/26046720818?pr=9723)

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .github/scripts/fitness-functions/rules/index.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { preventJavaScriptFileAdditions } from './javascript-additions';
```
Comment (legobeat) [inline .github/scripts/fitness-functions/rules/index.ts:1]: @sethkfman is there supposed to be a `javascript-additions.ts`..?

File: .github/scripts/fitness-functions/common/shared.ts [hunk 2]
```diff
@@ -19,6 +19,7 @@ function filterDiffByFilePath(diff: string, regex: RegExp): string {
         // corresponding diff block in
         .forEach((path) => {
           if (!regex.test(path)) {
+            console.log('PATH', path)
```
Comment (legobeat) [inline .github/scripts/fitness-functions/common/shared.ts:22]: Leftover `console.log`?

**Review by legobeat — DISMISSED** [review]
https://github.com/MetaMask/metamask-mobile/actions/runs/9199173846/job/25303545010?pr=9723

https://github.com/MetaMask/metamask-mobile/pull/9723/files#r1611184948

File: .github/scripts/fitness-functions/rules/index.ts (hunk 1 above)
Comment (sethkfman) [reply .github/scripts/fitness-functions/rules/index.ts:1]: I missed adding them

**Comment by sethkfman** [comment]
Troubleshooting fitness function issue

File: ios/MetaMask.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1487,7 +1487,11 @@
 				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
 				MTL_ENABLE_DEBUG_INFO = YES;
 				ONLY_ACTIVE_ARCH = YES;
-				OTHER_LDFLAGS = "$(inherited)";
+				OTHER_LDFLAGS = (
```
Comment (Cal-L) [inline ios/MetaMask.xcodeproj/project.pbxproj:1490]: Remove unrelated changes

File: .yarnrc [hunk 4]
```diff
@@ -1 +1 @@
-ignore-scripts true
+ignore-scripts true
```
Comment (Cal-L) [inline .yarnrc:1]: Was this accidentally commited?

**Review by Cal-L — COMMENTED** [review]
Left some comments

File: ios/MetaMask.xcodeproj/project.pbxproj (hunk 3 above)
Comment (sethkfman) [reply ios/MetaMask.xcodeproj/project.pbxproj:1490]: Removed

File: .yarnrc (hunk 4 above)
Comment (sethkfman) [reply .yarnrc:1]: Fixed

File: .github/scripts/fitness-functions/common/shared.ts (hunk 2 above)
Comment (sethkfman) [reply .github/scripts/fitness-functions/common/shared.ts]: removed

File: e2e/helpers copy.js
Comment (Cal-L) [inline e2e/helpers copy.js:1]: Was this file committed by accident?

Comment (sethkfman) [reply e2e/helpers copy.js:1]: removed test files

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**